pkg/store/bucket.go

```diff
@@ -876,16 +971,22 @@ func (s *BucketStore) Series(req *storepb.SeriesRequest, srv storepb.Store_Serie
 	req.MaxTime = s.limitMaxTime(req.MaxTime)
 
 	var (
-		ctx              = srv.Context()
-		stats            = &queryStats{}
-		res              []storepb.SeriesSet
-		mtx              sync.Mutex
-		g, gctx          = errgroup.WithContext(ctx)
-		resHints         = &hintspb.SeriesResponseHints{}
-		reqBlockMatchers []*labels.Matcher
-		chunksLimiter    = s.chunksLimiterFactory(s.metrics.queriesDropped)
+		ctx               = srv.Context()
+		stats             = &queryStats{}
+		res               []storepb.SeriesSet
+		mtx               sync.Mutex
+		g, gctx           = errgroup.WithContext(ctx)
+		resHints          = &hintspb.SeriesResponseHints{}
+		reqBlockMatchers  []*labels.Matcher
+		chunksLimiter     = s.chunksLimiterFactory(s.metrics.queriesDropped)
+		chunksSizeLimiter = s.chunksSizeLimiterFactory(s.metrics.queriesDropped)
 	)
 
+	chunksSizeLimiter, err = chunksSizeLimiter.NewWithFailedCounterFrom(chunksLimiter)
```

Comment:
Could you clarify why you need `NewWithFailedCounterFrom()`?

Reply:
@pracucci thanks for reviewing this.
Chunks limiter and chunks size limiter should have common `failedOnce` to avoid concurrent update of the `failedCounter`.

Reply:
more info on that: `store` does not know at the moment of `ChunksLimit` and `ChunksSizeLimit` creation what metrics will be used. Metrics are the part of the `BucketStore`. That is why store operates limiter factories and bucket creates new chunk size limiter with the sync.Once shared between limiters. Hope that helps

Reply:
I think `NewWithFailedCounterFrom()` is a bit overengineered.

There's no problem updating a metric concurrently, but we may want to distinguish the reason why a query was dropped. An option may be adding a "reason" label to `queriesDropped` and passing `s.metrics.queriesDropped.WithLabelValues("<reason>")` to both factories.